internal/cluster/cluster.go

```diff
@@ -20,54 +27,69 @@ func InstallHelm() {
 	cmd.Run()
 }
 
-func InitTerraform() {
-	currentDir := util.CurrentExecutableDir()
-	initScriptLocation := currentDir + "/terraform"
-	initScript := "./init.sh"
-
-	cmd := commandline.NewWithWorkingDir(initScript, []string{}, initScriptLocation)
-	cmd.Run()
+func InitTerraform(tf string, project string, env string) {
+	// Create Bucket if not present in the system
+	err := gcp.CreateGCSBucket(project, env)
+	if err != nil {
+		log.Fatal(err)
+	}
+	// Check if terraform initialized
+	if _, err := os.Stat(tf+"/.terraform"); os.IsNotExist(err) {
+		fmt.Printf("Terraform not Initialized: in %v: Intializong now...", tf+"/.terraform\n")
+		argsArray := util.StringTemplateToArgsArray("%s", "init")
+		cmd := commandline.NewWithWorkingDir("terraform", argsArray, tf)
+		cmd.Run()
+	}
 }
 
-//TODO: pass variables into the TF template/clustervars
-func CreateCluster(project string, env string, zone string) {
-	currentDir := util.CurrentExecutableDir()
-	tfFileLocation := currentDir + "/terraform/cluster"
-
-	varFileLoc := tfFileLocation + "/cluster.tfvars"
-	argsArray := util.StringTemplateToArgsArray("%s %s", "plan", "-var-file="+varFileLoc)
-	cmd := commandline.NewWithWorkingDir("terraform", argsArray, tfFileLocation)
+func PlanTerraform(tp string, tf string)  {
```

Comment:
that's the idea, map `plan, apply, ...` will add a small doc with the Python module I have to translate. It has everything in it to manage terraform but that should go in `terraform.go`

Reply:
Will implement it later. For now, I'll keep it. 

In the future, I'll consider using something like [this](https://github.com/getlunaform/go-terraform)

Reply:
Nice one you found it. Going to save so rewrites!!